### PR TITLE
Backports all needs for runTheMatrix to 80X

### PR DESCRIPTION
**Pull request by srimanob**
Backport all needs for MatrixInjector/runTheMatrix upto https://github.com/cms-sw/cmssw/pull/19681 (CMSSW_9_3_0_pre2) to fix the injection to reqmgr2

This is a proposal to fix https://hypernews.cern.ch/HyperNews/CMS/get/comp-ops/4412.html

Test of injection success,
https://hypernews.cern.ch/HyperNews/CMS/get/comp-ops/4412.html
@prebello

### CODE REVIEW

**Comment by srimanob** [comment]
Test of injection success,
srimanob_RVCMSSW_8_0_32ProdMinBias_13__gs7120p2_190301_125639_6718
@prebello

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py [hunk 1]
```diff
@@ -51,12 +51,19 @@ def runSelected(opt):
...
                      1001, #data+express
                      136.731, #2016B Photon data
+                     136.7611, #2016E JetHT reMINIAOD from 80X legacy
                      140.53, #2011 HI data
                      1330, #Run2 MC Zmm
-                     135.4 #Run 2 Zee ttbar
+                     135.4, #Run 2 Zee ttbar
+                     10042.0, #2017 ZMM
+                     10024.0, #2017 ttbar
+                     10824.0, #2018 ttbar
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: workflows beyond 2017 do not exist in 80X, remove this line and the following two lines

File: Configuration/PyReleaseValidation/python/MatrixInjector.py [hunk 2]
```diff
@@ -185,24 +187,41 @@ def prepare(self,mReader, directories, mode='init'):
...
+            wmsplit['DIGIUP15']=1 
+            wmsplit['RECOUP15']=1 
             wmsplit['RECOAODUP15']=5
             wmsplit['DBLMINIAODMCUP15NODQM']=5
+            wmsplit['DigiFull']=5
+            wmsplit['RecoFull']=5
+            wmsplit['DigiFullPU']=1
+            wmsplit['RecoFullPU']=1
+            wmsplit['RECOHID11']=1
+            wmsplit['DigiFullPU_2023D17PU']=1
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/MatrixInjector.py]: probably these 2023-specific steps can be removed too

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py (hunk 1 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: Thx, done. Please check.

File: Configuration/PyReleaseValidation/python/MatrixInjector.py (hunk 2 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixInjector.py]: Thx, done. Please check.

**Comment by fabiocos** [comment]
please test

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py [hunk 3]
```diff
@@ -132,6 +139,12 @@ def runSelected(opt):
                       dest='step1Only',
                       default=False
                       )
+    parser.add_option('--maxSteps',
```
Comment (fabiocos) [inline Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: @srimanob it looks to me there is no code to consume this option in 8_0_X

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py [hunk 4]
```diff
@@ -51,12 +51,13 @@ def runSelected(opt):
                      8, #BH/Cosmic MC
                      25, #MC ttbar
                      4.22, #cosmic data
+                     4.53, #run1 data + miniAOD
                      1000, #data+prompt
                      1001, #data+express
                      136.731, #2016B Photon data
+                     136.7611, #2016E JetHT reMINIAOD from 80X legacy
```
Comment (fabiocos) [inline Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: as this PR is now, this does not work, not sure whether we really need for the purpose of this PR

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py (hunk 3 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: @fabiocos I removed it as suggestion.

File: Configuration/PyReleaseValidation/scripts/runTheMatrix.py (hunk 4 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/scripts/runTheMatrix.py]: @fabiocos removed

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
@kpedro88 @prebello @zhenhu any other comment?

**Comment by prebello** [comment]
none from my side. Thank you.

**Comment by prebello** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
+1

**Pull request merged**